src/Umbraco.Core/Persistence/Repositories/Implement/DocumentRepository.cs

```diff
@@ -266,8 +266,8 @@ protected override void PersistNewItem(IContent entity)
             var publishing = content.PublishedState == PublishedState.Publishing;
 
             // ensure that the default template is assigned
-            if (entity.Template == null)
-                entity.Template = entity.ContentType.DefaultTemplate;
+            if (entity.TemplateId == 0)
```

Comment:
You should be able to handle it like the nullable it is:

`if (entity.TemplateId == null)
                entity.Template = entity.ContentType.DefaultTemplate;	                entity.TemplateId = entity.ContentType.DefaultTemplate?.Id;`